### ISSUE TITLE
How to create image-parse images

### ISSUE THREAD

**Issue by furkan59** [sangyun884/HR-VITON#76]
I have human images but how can I create image-parse images in folder "VITON-HD\data\train\image-parse-v3".

**Comment by KIRAN-MK-211** [comment]
https://colab.research.google.com/drive/1JOwOPaChoc9GzyBi5FUEYTSaP2qxJl10?usp=sharing#scrollTo=VtZDKLL8jWLr 
You can refer this notebook. results are not exactly same as shown in the paper, but works well.

**Comment by garychan22** [comment]
I think the problem is how to get the exact training data of VITON-HD according to the documentation

> I saved a parse map image using PIL.Image with P mode. The color of the parse map image in our dataset(VITON-HD) is just for the visualization, it has 0~19 uint values.

Should we just change this line 
[https://github.com/Engineering-Course/CIHP_PGN/blob/0cf1cbe54a44fc86abe2023b0e762df3f9605242/test_pgn.py#L194C11-L194C11](url)
```
cv2.imwrite('{}/{}.png'.format(parsing_dir, img_id), parsing_[0,:,:,0])
```
to
```
Image.fromarray(parsing_[0,:,:,0]).convert("P").save('{}/{}.png'.format(parsing_dir, img_id))
```

**Comment by furkan59** [comment]
This one seems fine thanks 
https://colab.research.google.com/drive/1JOwOPaChoc9GzyBi5FUEYTSaP2qxJl10?usp=sharing#scrollTo=VtZDKLL8jWLr

**Comment by garychan22** [comment]
but the results produced by this https://colab.research.google.com/drive/1JOwOPaChoc9GzyBi5FUEYTSaP2qxJl10?usp=sharing#scrollTo=VtZDKLL8jWLr are far from the ones in viton regardless of the color

example here 

the one in viton
![00000_00](https://github.com/sangyun884/HR-VITON/assets/108175311/04e2d4f0-b41c-407e-bd4f-ebc8925a6fd5)

the one produced by the referred package, **where the torso is missing** and mistakenly classified into background
![00000_00](https://github.com/sangyun884/HR-VITON/assets/108175311/0b19881e-b7ba-4db1-9786-21c82084f211)

**Comment by triminh12042002** [comment]
As the processing part of the HR-VITON code: https://github.com/sangyun884/HR-VITON/blob/main/Preprocessing.md

> Check https://github.com/Engineering-Course/CIHP_PGN for human parsing. I inferenced a parse map on 256x192 resolution, and upsample it to 1024x768. Then you can see that it has a alias artifact, so I smooth it using "torchgeometry.image.GaussianBlur((15, 15), (3, 3))". I saved a parse map image using PIL.Image with P mode. The color of the parse map image in our dataset(VITON-HD) is just for the visualization, it has 0~19 uint values.

Using repo https://github.com/Engineering-Course/CIHP_PGN
and with the human image you can use the test_pgn.py to generate the results with the trained models.
Before it you  must setup include: 
1. Install requirements package, you should use virtual env with python3.7 like conda to install these package:
  a. pip install -r requirements.pip
  b. pip install --upgrade pip
  c. pip install protobuf==3.20.1
d.  run the command: python3 test_pgn.py (to make sure your code already run first)
2. Add your image to folder image (datasets/CIHP/image) 
3. copy 1 test image, which already exist in folder labels, resize it to the size of your image, and change the name to your image  filename to make sure code don't fail (may be for validation purpose) 
4. Add that new resized image to both folder labels and edge
the result may look like:
![Screenshot from 2023-12-20 20-49-54](https://github.com/sangyun884/HR-VITON/assets/80403992/25e01d1c-40b0-401d-bbce-56b0a1157a03)
I have duplicate and resize 0032190.png to size of my Ronaldo1.png (using online tool), and copy it to edges and labels folder
5. Add your new image name and path to file /list/val_id.txt and val.txt
6. Adjust the code in file `test_pgn.py`  to store the result image with "P" mode: `parsing_im = parsing_im.convert('P')`
before the line 198 `parsing_im.save('{}/{}_vis.png'.format(parsing_dir, img_id))`

Now run the command: python3 test_pgn.py
The output image path will be: segmentation/CIHP_PGN/output/cihp_parsing_maps/Ronaldo1_vis.png
This result will be use for the image-parse-v3 folder of VITON - HD dataset

**Comment by whiterose199187** [comment]
hi @triminh12042002 

I tries this approach but looks like that repo is using pretty old versions of tensorflow. How did you get it to work? Did you use python 2.x ?

**Comment by triminh12042002** [comment]
No, I'm using python 3.7
I think u should use conda to create an env with python3.7 and install the requirements package as step 1

**Comment by kapoorkhushi03** [comment]
@whiterose199187 were u able to implement viton-HD?

**Comment by vahid037** [comment]
> @whiterose199187 were u able to implement viton-HD?

you should implement vton-hd and preprocessing data in different virtual environment.
